### ISSUE TITLE
Warning when using variable in `Literal` defintion

### ISSUE THREAD

**Issue by joellidin** [microsoft/pyright#7770]
**Describe the bug**
I am trying to have a Literal defined from a string literal and I get a warning saying:
```console
Diagnostics:
1. Pyright: Type arguments for "Literal" must be None, a literal value (int, bool, str, or bytes), or an enum value
```
I am not sure this is expected behavior bur the message says it expects a literal value which is what I provide it with.

**Code or Screenshots**
The warning can be reproduced easily by the code below:

```python
from typing import Literal
from dataclasses import dataclass

baz = "hola"


@dataclass
class Foo:
    bar: Literal[baz] = baz
```
See this print screen:
![image](https://github.com/microsoft/pyright/assets/19657866/04f48df0-3b44-4e82-bc8c-31006f077976)

**VS Code extension or command-line**
I am using pyright within neovim an my installed version is `1.1.358`.

**Comment by erictraut** [comment]
This is definitely expected behavior. The use of variables within a static type annotation is expressly forbidden by the Python typing specification — and for good reason. The correct annotation in this case is `Literal["hola"]`.

**Comment by joellidin** [comment]
Thank you!
